nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/BlobStatus.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Florent Guillaume
+ */
+package org.nuxeo.ecm.core.blob;
+
+import java.time.Instant;
+
+/**
+ * Status associated to a blob in storage.
+ *
+ * @since 11.1
+ */
+public class BlobStatus {
+
+    protected String storageClass;
+
+    protected boolean downloadable = true;
+
+    protected Instant downloadableUntil;
+
+    public BlobStatus withStorageClass(String storageClass) {
+        this.storageClass = storageClass;
+        return this;
+    }
+
+    public BlobStatus withDownloadable(boolean downloadable) {
+        this.downloadable = downloadable;
+        return this;
+    }
+
+    public BlobStatus withDownloadableUntil(Instant downloadableUntil) {
+        this.downloadableUntil = downloadableUntil;
+        return this;
+    }
+
+    /**
+     * The storage class, or {@code null} for the standard storage class.
+     */
+    public String getStorageClass() {
+        return storageClass;
+    }
```

Comment:
what if we hide the fact that `null` is a the Standard storage as it is low level implementation detail (mainly AWS for instant), i think your idea is to use the blob status like more generic, but `...or {@code null} for the standard storage class.` is mainly an AWS implementation.
or to be more generic perhaps we shouldn't give this information and replace it by more generic term `storageDetails`

Reply:
I'll change so that the default storage class is always represented as `null`, even when calling `getStatus` (so if the storage returns a value that corresponds to its standard storage class I'll turn it into null). That helps to build a "standard" availability with just `new BlobStatus()`